CONTRIBUTING.md

```diff
@@ -175,3 +175,90 @@ For testing and development you can find self-signed certificates under
 ```bash
  openssl req -nodes -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -sha256 -days 365 -subj '/CN=localhost'
 ```
+
+## Logging Principles
+Logging plays a crucial role for administrators, users and developers of our tool, which is why it
+is important to have certain standards in regards on when and how to use which type of logger.
+
+The following sections will outline
+- what information should be logged.
+- when information should be logged.
+- how information should be logged.
+
+### What information should be logged.
+#### Unknown-Error Log Message
+These types of messages are things such as an unexpected exception was thrown, which should
+provider relative information and context on the exception.
+The following questions should be answered with these log messages.
+- What error happened
+- what data was used to cause this error
+- when did the error happen
+- ...
+
+
+#### Informational Log Message
+These types of log messages are used to give application administrators contexts to user errors,
```

Comment:
Should we let JAX-RS log all incoming requests? If so, how do we configure it to comply with our requirements?

Reply:
I personally don't like logging info level logs by default. At least not info logs for core parts of the application. I feel like it clutters more than add context.

My opinion towards this isn't very strong though, so if you consider it helpful i'll gladly add the functionality.